otx/core/data/adapter/base_dataset_adapter.py

```diff
@@ -0,0 +1,250 @@
+        val_data_roots (str): Path for validation data
+        test_data_roots (str): Path for test data
+        unlabeled_data_roots (str): Path for unlabeled data
+
+    """
+
+    def __init__(
+        self,
+        task_type: TaskType,
+        train_data_roots: str = None,
+        val_data_roots: str = None,
+        test_data_roots: str = None,
+        unlabeled_data_roots: str = None,
+    ):
+        self.task_type = task_type
+        self.domain = task_type.domain
+        self.data_type = None  # type: Any
+        self.is_train_phase = None  # type: Any
+
+        self.dataset = self._import_dataset(
+            train_data_roots=train_data_roots,
+            val_data_roots=val_data_roots,
+            test_data_roots=test_data_roots,
+            unlabeled_data_roots=unlabeled_data_roots,
+        )
+
+        self.category_items = None  # type: Any
+        self.label_groups = None  # type: Any
+        self.label_entities = None  # type: Any
+        self.label_schema = None  # type: Union[LabelSchemaEntity, None]
+
+    def _import_dataset(
+        self,
+        train_data_roots: str = None,
+        val_data_roots: str = None,
+        test_data_roots: str = None,
+        unlabeled_data_roots: str = None,
+    ) -> Dict[Subset, DatumaroDataset]:
+        """Import dataset by using Datumaro.import_from() method.
+
+        Args:
+            train_data_roots (str): Path for training data
+            val_data_roots (str): Path for validation data
+            test_data_roots (str): Path for test data
+            unlabeled_data_roots (str): Path for unlabeled data
+            unlabeled_file_lists (str): Path for unlabeled file list
+
+        Returns:
+            DatumaroDataset: Datumaro Dataset
+        """
+        dataset = {}
+
+        # Construct dataset for training, validation, testing, unlabeled
+        if train_data_roots:
+            # Find self.data_type and task_type
+            data_type_candidates = self._detect_dataset_format(path=train_data_roots)
+            self.data_type = self._select_data_type(data_type_candidates)
+
+            datumaro_dataset = DatumaroDataset.import_from(train_data_roots, format=self.data_type)
+
+            # Prepare subsets by using Datumaro dataset
+            for k, v in datumaro_dataset.subsets().items():
+                if "train" in k or "default" in k:
+                    dataset[Subset.TRAINING] = v
+                elif "val" in k:
+                    dataset[Subset.VALIDATION] = v
+            self.is_train_phase = True
+
+            # If validation is manually defined --> set the validation data according to user's input
+            if val_data_roots:
+                val_data_candidates = self._detect_dataset_format(path=val_data_roots)
+                val_data_type = self._select_data_type(val_data_candidates)
+                dataset[Subset.VALIDATION] = DatumaroDataset.import_from(val_data_roots, format=val_data_type)
```

Comment:
Is it overwriting validation dataset defined by user in cli? Then auto-split by datumaro import is not used for validation dataset. Then the validation dataset previously made is needed to be merged in training.

Reply:
Could you explain why you think that this line overwrites the validation dataset?
I think this line is the first line that defines the validation dataset

Reply:
at L109, didn't it make validation dataset by datumaro auto-split from train root?

Reply:
For specific datasets (COCO, VOC), Datumaro can make a validation dataset by default when COCO dataset also includes the validation annotation file. However, if the user wants to use a different validation dataset, we need to overwrite the validation dataset. 
(i.e. if COCO dataset includes `instances_train.json` and `instances_val.json`, then Datumaro automatically import `train` and `validation` dataset by using two json files. However, if the user wants to use `instances_val_fish.json`, then we should change the previous validation set. And in this case, we don't need to convert `instances_val.json` to train dataset.)

And for ImageNet, Image-Mask pair dataset, Datumaro can't make a validation dataset automatically. So in this case, this is the first time to make a validation dataset.

In the current implementation, Auto-split is not implemented. So, I attached `TODO` mark on that.